graphistry/compute/cluster.py

```diff
@@ -0,0 +1,391 @@
+
+        """
+
+        res = self.bind()
+        res = res._cluster_dbscan(
+            res,
+            kind=kind,
+            cols=cols,
+            fit_umap_embedding=fit_umap_embedding,
+            target=target,
+            min_dist=min_dist,
+            min_samples=min_samples,
+            verbose=verbose,
+            *args,
+            **kwargs,
+        )
+        #res = res.encode_point_color(column=DBSCAN, as_categorical=True)
+
+        return res
+
+    def _transform_dbscan(
+        self, df: pd.DataFrame, ydf, kind, verbose
+    ) -> Tuple[Union[pd.DataFrame, None], pd.DataFrame, pd.DataFrame, pd.DataFrame]:
+        
+        res = self.bind()
+        if hasattr(res, "_dbscan_params"):
+            # Assume that we are transforming to last fit of dbscan
+            cols = res._dbscan_params["cols"]
+            umap = res._dbscan_params["fit_umap_embedding"]
+            target = res._dbscan_params["target"]
+
+            dbscan = res._node_dbscan if kind == "nodes" else res._edge_dbscan
+
+            emb = None
+            if umap and cols is None:
+                emb, X, y = res.transform_umap(df, ydf, kind=kind, return_graph=False)
+            else:
+                X, y = res.transform(df, ydf, kind=kind, return_graph=False)
+            XX = X
+            if target:
+                XX = y
+            if cols is not None:
+                XX = get_matrix_by_column_parts(XX, cols)
+
+            if umap:
+                X_ = emb
+            else:
+                X_ = XX
+
+            labels = dbscan_predict(X_, dbscan)  # type: ignore
+            if umap and cols is None:
+                df = df.assign(_dbscan=labels, x=emb.x, y=emb.y)  # type: ignore
+            else:
+                df = df.assign(_dbscan=labels)
+            
+            if verbose:
+                print(f"Transformed DBSCAN: {len(df[DBSCAN].unique())} clusters")
+
+            return emb, X, y, df  # type: ignore
+        else:
+            raise Exception("No dbscan model found. Please run `g.dbscan()` first")
+
+    def transform_dbscan(
+        self,
+        df: pd.DataFrame,
+        y: Optional[pd.DataFrame] = None,
+        min_dist: Union[float, str] = "auto",
+        infer_umap_embedding: bool = False,
+        sample: Optional[int] = None,
+        n_neighbors: Optional[int] = None,
+        kind: str = "nodes",
+        return_graph: bool = True,
+        verbose: bool = False,
+        ):  # type: ignore
+        """
+        Transforms a minibatch dataframe to one with a new column '_dbscan' containing the DBSCAN cluster 
+        labels on the minibatch and generates a graph with the minibatch and the original graph, with edges 
+        between the minibatch and the original graph inferred from the umap embedding or features dataframe.
+        Graph nodes | edges will be colored by '_dbscan' column.
+            
+            Examples:
```

Comment:
@silkspace I don't think these will get picked up as valid ~rst for the read the docs, see folder `docs/README` for local testing etc

Reply:
Will do